distributed/semaphore.py

```diff
@@ -412,7 +412,7 @@ def __init__(
 
     @property
     def asynchronous(self):
-        return self.loop is IOLoop.current()
+        return self.loop is IOLoop.current(instance=False)
```

Comment:
`instance` arg is [deprecated in tornado 5](https://www.tornadoweb.org/en/stable/ioloop.html#tornado.ioloop.IOLoop.current), and minimum supported tornado is 6.0.3.